source/utilities.cc

```diff
@@ -104,6 +105,43 @@ namespace aspect
       return static_cast<bool>(ifile); // only in c++11 you can convert to bool directly
     }
 
+    void
+    load_and_distribute_file_content(const std::string &filename,
+                                     const MPI_Comm &comm,
+                                     std::stringstream &datastream)
+    {
+      // Reset datastream
+      datastream.str(std::string());
+
+      if (Utilities::MPI::this_mpi_process(comm) == 0)
+        {
+          std::ifstream filestream(filename);
+          AssertThrow (filestream,
+                       ExcMessage (std::string("Could not open file <") + filename + ">."));
+
+          // Read data from disk
+          filestream >> datastream.rdbuf();
+          std::string data_string = datastream.str();
+          unsigned int filesize = data_string.size();
+
+          // Distribute data_size and data across processes
+          MPI_Bcast(&filesize,1,MPI_UNSIGNED,0,comm);
+          MPI_Bcast(&data_string[0],filesize,MPI_CHAR,0,comm);
```

Comment:
I also thought so, but MPI_Probe does not work with Bcast, because Bcast is a blocking collective operation that will only start when all processes have called it (so no process should wait with a MPI_Probe for the Bcast to arrive).